crates/payload/builder/src/lib.rs

```diff
@@ -153,38 +149,49 @@ impl<Provider: ChainSpecProvider> TempoPayloadBuilder<Provider> {
     /// Builds system transactions to seal the block.
     ///
     /// Returns a vector of system transactions that must be executed at the end of each block:
-    /// 1. Fee manager executeBlock - processes collected fees
+    /// 1. Fee manager executeBlock - processes collected fees (pre-AllegroModerato only)
     /// 2. Stablecoin exchange executeBlock - commits pending orders
     /// 3. Subblocks signatures - validates subblock signatures
     fn build_seal_block_txs(
         &self,
-        block_env: &BlockEnv,
+        evm: &TempoEvm<impl Database>,
         subblocks: &[RecoveredSubBlock],
     ) -> Vec<Recovered<TempoTxEnvelope>> {
         let chain_id = Some(self.provider.chain_spec().chain().id());
+        let block_env = evm.block();
+        let spec = &evm.ctx().cfg.spec;
+
+        let mut txs = Vec::with_capacity(3);
```

Comment:
nit:
```suggestion
        let mut txs = Vec::with_capacity(if spec.is_allegro_moderato() { 2 } else { 3 });
```

note: may require fmt